### ISSUE TITLE
ancient upgrade :interactive is updating the wrong dependencies in project.clj

### ISSUE THREAD

**Issue by nchurch** [xsc/lein-ancient#41]
I tried upgrade :interactive like so on my project.clj:

```
[org.clojure/data.xml "0.0.8"] is available but we use "0.0.6"
Do you want to upgrade? [yes/no] yes

[org.clojure/core.incubator "0.1.3"] is available but we use "0.1.2"
Do you want to upgrade? [yes/no] yes

[hiccup "1.0.5"] is available but we use "1.0.1"
Do you want to upgrade? [yes/no] yes

[hickory "0.5.4"] is available but we use "0.2.0"
Do you want to upgrade? [yes/no] yes

[seesaw "1.4.4"] is available but we use "1.4.2"
Do you want to upgrade? [yes/no] yes

[org.cloudhoist/stevedore "0.8.0-alpha.1"] is available but we use "0.7.2"
Do you want to upgrade? [yes/no] no

[org.cloudhoist/pallet "0.8.0-alpha.9"] is available but we use "0.7.2"
Do you want to upgrade? [yes/no] no

[hiccup-bridge "1.0.1"] is available but we use "1.0.0-SNAPSHOT"
Do you want to upgrade? [yes/no] yes

[prismatic/plumbing "0.3.5"] is available but we use "0.3.3"
Do you want to upgrade? [yes/no] yes
```

For some of the dependencies it failed to update; for others it swapped version numbers.  Here are some results in the project.clj file:

```
[org.clojure/core.incubator "0.1.2"]  -- failed to upgrade
[org.cloudhoist/pallet "1.0.1"]  -- swapped with the new hiccup-bridge version number evidently
[hiccup-bridge "0.3.5"] -- swapped with the new prismatic plumbing version number evidently
[prismatic/plumbing "0.3.3"] -- failed to upgrade
```

The mangled project.clj is below:

```clojure
(defproject my-project/my-project "0.1.0-SNAPSHOT" 
  :license {:name "Eclipse Public License",
            :url "http://www.eclipse.org/legal/epl-v10.html"}
  :dependencies [[org.clojure/clojure "1.6.0"]
                 [org.clojure/data.xml "0.0.8"]
                 [org.clojure/core.incubator "0.1.3"]
                 [clj-tagsoup "0.3.0"]
                 [hiccup "1.0.5"]
                 [hickory "0.5.4"]
                 [seesaw "1.4.4"]
                 [org.clojure/core.incubator "0.1.2"]
                 [org.cloudhoist/stevedore "0.7.2"]
                 [org.cloudhoist/pallet "1.0.1"]
                 [hiccup-bridge "0.3.5"]
                 [prismatic/plumbing "0.3.3"]
                 [utils "1.0.0-SNAPSHOT"]]
  :url "http://example.com/FIXME"
  :repositories {"sonatype"
                 "http://oss.sonatype.org/content/repositories/releases"}
  :min-lein-version "2.0.0"
  :description "FIXME: write description")
```

**Comment by xsc** [comment]
This should be fixed. Can you try the latest SNAPSHOT?

```clojure
```

**Comment by nchurch** [comment]
This worked great even with the duplicated dependency (I checked out the project as it was when this bug showed up).

(I had to use the :no-tests option as I only had the default testing, which has an intentionally failing test.  The error message from that btw was as follows:
running test task ["test"] ...
(error) tests have failed (use ':no-tests' to disable testing).
(error) #<NullPointerException java.lang.NullPointerException>)
